web/thesauruses/lua/5/strings.json

```diff
@@ -0,0 +1,272 @@
+        "not-implemented": true
+      },
+      "decode_from_base64": {
+        "name": "Decode string from Base64 format",
+        "not-implemented": true
+      },
+      "format_string_function": {
+        "name": "Function to format a string",
+        "code": "tool = 'codethesaurus'\nformated = string.format('The great learning tool %s',tool)\nprint(formated)"
+      },
+      "parameter_format_in_order": {
+        "name": "Parameter used in format function if they're used in order",
+        "code": "string.format(formatstring, ...)"
+      },
+      "parameter_format_numerical": {
+        "name": "Parameter used in format function if they're numerically numbered",
+        "not-implemented": true
+      },
+      "parameter_format_by_name": {
+        "name": "Paramater used in format function if they're named variables",
+        "not-implemented": true
+      },
+      "format_as_integer": {
+        "name": "Format parameter as an integer",
+        "code": "str = '12'\nprint(string.format('%d',str))"
+      },
+      "format_as_decimal": {
+        "name": "Format parameter as a decimal number",
+        "not-implemented": true
+      },
+      "format_as_fixed_decimal": {
+        "name": "Format parameter as a fixed-point decimal number",
+        "code": "str = '12.53232'\nprint(string.format('%0.2f',str))",
+        "comment":"Output: 12.53\n%0.[N]f [where N is how many decimal you want after (.)point]"
+      },
+      "format_as_currency": {
+        "name": "Format parameter as a currency number",
+        "not-implemented": true
+      },
+      "format_as_percentage": {
+        "name": "Format parameter as a percentage number",
+        "not-implemented": true
+      },
+      "format_number_with_separators": {
+        "name": "Format number with thousand separators",
+        "not-implemented": true
+      },
+      "format_number_with_positive_negative_sign": {
+        "name": "Format number with positive/negative signs",
+        "not-implemented": true
+      },
+      "format_number_in_scientific_notation_little_e": {
+        "name": "Format number with scientific notation with 'e'",
+        "code": "str = '12645893253232'\nprint(string.format('%e',str))"
+      },
+      "format_number_in_scientific_notation_big_e": {
+        "name": "Format number with scientific notation with 'E'",
+        "code": "str = '12645893253232'\nprint(string.format('%E',str))"
+      },
+      "format_number_in_binary": {
+        "name": "Format number into binary",
+        "not-implemented": true
+      },
+      "format_number_in_octal": {
+        "name": "Format number into octal",
+        "code": "str = '8'\nprint(string.format('%o',str))"
+      },
+      "format_number_in_hexadecimal": {
+        "name": "Format number into hexadecimal",
+        "code": "str = '10'\nprint(string.format('%x',str))",
+        "comment":"%x for lowercase letters.\n%X upper letters."
+      }

```

Comment:
Same syntax will also work for number !

Reply:
You know that and I do but someone new to the language might not!
Since these concepts are about converting numbers to strings and not strings to strings the examples should use numbers imo